### PR TITLE
Avoid eager task initialization

### LINKED ISSUE #201: Please create Bugsnag tasks using Gradle's creation avoidance API
### Expected behavior
Create bugsnag tasks lazily using this API
https://docs.gradle.org/current/userguide/task_configuration_avoidance.html
Rather than [creating them](https://github.com/bugsnag/bugsnag-android-gradle-plugin/blob/master/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy#L125) you can [register them](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_general) 

### Observed behavior
Plugin should lazily create tasks to avoid hitting configuration time.
Our build scan shows that the bugsnag plugin is creating 921 task at configuration time. However, most of them are for variants not related to the task running so I believe they could be registered instead of created as shared above.
```
com.bugsnag.android.gradle | Plugin | 0.186s |   | Applied to 1 project | 916 tasks created
```

### Steps to reproduce
Run a build `--scan` https://guides.gradle.org/creating-build-scans/
and go to Performance > configuration > search for `com.bugsnag.android.gradle`
For us it creates 921 tasks the creation can be deferred to when they are actually needed

### Version
4.22.3

### Additional information
*[Insert any additional information]*

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #201 by avoiding eagerly initializing other tasks.

Still needs more work but sticking this up for early feedback. Will continue iterating on this! Commits get a little messy due to things being interconnected :|

### CODE REVIEW

File: src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt [hunk 1]
```diff
@@ -170,48 +170,44 @@ class BugsnagPlugin : Plugin<Project> {
...
-            symbolFileTask?.get()?.let { task ->
-                val ndkSearchDirs = symbolFileTask.get().searchDirectories
-                releasesTask.get().ndkMappingFileProperty.from(ndkSearchDirs)
-                task.uploadRequestClient.set(ndkUploadClientProvider)
-            }
-            releasesTask.get().uploadRequestClient.set(releasesUploadClientProvider)
-            releasesTask.get().manifestInfoFile.set(manifestInfoFile)
-            symbolFileTask?.get()?.manifestInfoFile?.set(manifestInfoFile)
-            releasesTask.get().manifestInfoFile.set(manifestInfoFile)
+            }?.registerWithAssembleTasks(variant, output, bugsnag, project.provider { true })
```
Comment (fractalwrench) [inline src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt:199]: I think this should probably be `project.provider { bugsnag.uploadNdkMappings }`

File: src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt [hunk 2]
```diff
@@ -283,11 +284,12 @@ class BugsnagPlugin : Plugin<Project> {
         return BugsnagUploadNdkTask.register(project, taskName) {
             this.requestOutputFile.set(requestOutputFile)
             projectRoot.set(bugsnag.projectRoot.getOrElse(project.projectDir.toString()))
-            searchDirectories.from(getSearchDirectories(project, variant))
+            searchDirectories.from()
```
[the remark this answers is not in the document]
Comment (ZacSweers) [reply src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt:287]: accident 😬

File: src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt (hunk 1 above)
Comment (ZacSweers) [reply src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt:199]: Good call. It was hardcoded to `true` before so I was just matching this originally

**Comment by phillipsam** [comment]
This has been released in [v5.0.0](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v5.0.0)

**Pull request merged**